lib/src/grpc/mock/mod.rs

```diff
@@ -0,0 +1,264 @@
+#![doc = include_str!("README.md")]
+
+use std::convert::Infallible;
+use tokio::sync::oneshot::{self, Sender};
+use tonic::body::BoxBody;
+use tonic::transport::{NamedService, Server};
+use tower::Service;
+
+/// Starts a [`tokio`] server listening on 50051 for our `MockService`
+/// # Example
+/// ```
+/// use lib_common::grpc::mock::start_server;
+///
+/// pub mod mock_server {
+///     #![allow(unused_qualifications)]
+///     include!("grpc_server.rs");
+/// }
+/// use mock_server::server::{MockService, MockServer};
+/// use mock_server::*;
+///
+/// async fn start() {
+///     let service = GrpcMockImpl::default();
+///     let shutdown_tx = start_server("0.0.0.0:50051", MockServer::new(service)).await.expect("Could not start server.");
+///     // send server shutdown signal
+///     shutdown_tx.send(()).expect("Unable to shutdown server");
+/// }
+///
+/// #[derive(Default, Debug, Clone, Copy)]
+/// pub struct GrpcMockImpl {}
+///
+/// #[tonic::async_trait]
+/// impl MockService for GrpcMockImpl {
+///     async fn is_ready(
+///         &self,
+///         request: tonic::Request<ReadyRequest>,
+///     ) -> Result<tonic::Response<ReadyResponse>, tonic::Status> {
+///         println!("Got a request: {:?}", request);
+///         let reply = ReadyResponse { ready: true };
+///         Ok(tonic::Response::new(reply))
+///     }
+/// }
+/// ```
+pub async fn start_server<S>(
+    addr: &str,
+    service: S,
+) -> Result<Sender<()>, Box<dyn std::error::Error>>
+where
+    S: Service<http::Request<hyper::Body>, Response = http::Response<BoxBody>, Error = Infallible>
+        + NamedService
+        + Clone
+        + Send
+        + 'static,
+    S::Future: Send + 'static,
+{
+    println!("Starting server on {}", addr);
```

Comment:
a few of the print statements in this file could be preceded by '(fn_name)'

Reply:
https://github.com/Arrow-air/lib-common/pull/12/commits/2e66ec06357b84e3c7a2e9612408b6e3f3f39e9c